soh/soh/Enhancements/randomizer/Plandomizer.cpp

```diff
@@ -0,0 +1,1050 @@
+    hintTootip = 
+        "The following options are available:\n"
+        "- Use \\n to create New Lines.\n"
+        "- Use %g to change the text color to Green,\n"
+        " - %r for Red, %y for Yellow, and %w for White\n"
+        "   can also be used as color examples.";
+
+    return hintTootip;
+}
+
+std::string extractNumberInParentheses(const std::string& text) {
+    size_t start = text.find('(');
+    size_t end = text.find(')');
+
+    if (start != std::string::npos && end != std::string::npos && start < end) {
+        return text.substr(start + 1, end - start - 1);
+    }
+    return "";
+}
+
+void PlandomizerPopulateSeedList() {
+    existingSeedList.clear();
+    auto spoilerPath = Ship::Context::GetPathRelativeToAppDirectory("Randomizer");
+
+    for (const auto& entry : std::filesystem::directory_iterator(spoilerPath)) {
+        if (entry.is_regular_file() && entry.path().extension() == ".json") {
+            existingSeedList.push_back(entry.path().stem().string());
+        }
+    }
+}
+
+void PlandomizerItemImageCorrection(Rando::Item randoItem) {
+    textureID = 0;
+    textureUV0 = ImVec2( 0, 0 );
+    textureUV1 = ImVec2( 1, 1 );
+
+    itemColor = plandomizerGetItemColor(randoItem);
+
+    if (randoItem.GetItemType() == ITEMTYPE_SMALLKEY || randoItem.GetItemType() == ITEMTYPE_FORTRESS_SMALLKEY) {
+        textureID = Ship::Context::GetInstance()->GetWindow()->GetGui()->GetTextureByName("ITEM_KEY_SMALL");
+        return;
+    }
+    if (randoItem.GetItemType() == ITEMTYPE_BOSSKEY) {
+        textureID = Ship::Context::GetInstance()->GetWindow()->GetGui()->GetTextureByName("ITEM_KEY_BOSS");
+        return;
+    }
+
+    for (auto& map : itemImageMap) {
+        if (map.first == randoItem.GetRandomizerGet()) {
+            textureID = Ship::Context::GetInstance()->GetWindow()->GetGui()->GetTextureByName(map.second.c_str());
+            if (map.second.find("ITEM_ARROWS") != std::string::npos) {
+                textureUV0 = ImVec2( 0, 1 );
+                textureUV1 = ImVec2( 1, 0 );
+            }
+            if (map.second.find("ITEM_TRIFORCE") != std::string::npos) {
+                textureUV0 = ImVec2( 1, 1 );
+                textureUV1 = ImVec2( 0, 0 );
+            }
+            break;
+        }
+    }
+
+    if (randoItem.GetRandomizerGet() >= RG_GOHMA_SOUL && randoItem.GetRandomizerGet() <= RG_GANON_SOUL) {
+        textureID = Ship::Context::GetInstance()->GetWindow()->GetGui()->GetTextureByName("BOSS_SOUL");
+    }
+
+    if (textureID == 0) {
+        textureID = Ship::Context::GetInstance()->GetWindow()->GetGui()->GetTextureByName(itemMapping[randoItem.GetGIEntry()->itemId].name);
+    }
+}
+
+void PlandomizerRandomizeHint(int32_t status, int32_t index) {
+    int32_t roll;
+    if (status == HINT_SINGLE) {
+        roll = (rand() % Rando::StaticData::hintTextTable.size());
+        plandoHintData[index].hintText = Rando::StaticData::hintTextTable[roll].GetHintMessage().GetForCurrentLanguage();
```

Comment:
Rolling any hint text is unwise, as there's quite a few that won't be complete hints. GetRandomJunkHint in Hints.cpp gives you a random junk hint key, which is what this code seemed to be going for, though it will need editing to end the range on RHT_JUNK_CREW_29 instead of RHT_JUNK_SG_8.

Whether these are good junk hints is another question, but IMO this was already an issue with the old usage of them as the main plandomiser hint choices, so at least handling them as junk hints avoids more special cases and the associated tech debt.